src/Symfony/Component/Workflow/Registry.php

```diff
@@ -56,16 +63,19 @@ public function get($subject, $workflowName = null)
         return $matched;
     }
 
-    private function supports(Workflow $workflow, $className, $subject, $name)
+    private function supports(Workflow $workflow, $supportStrategy, $subject, $workflowName)
     {
-        if (!$subject instanceof $className) {
+        if (is_string($supportStrategy) && !$subject instanceof $supportStrategy) {
```

Comment:
Rather than dealing with BC here, I would convert the string to a SupportStrategy when setting it instead.